Objects/object.c

```diff
@@ -1420,6 +1431,7 @@ _PyObject_GenericSetAttrWithDict(PyObject *obj, PyObject *name,
             res = PyDict_SetItem(dict, name, value);
         Py_DECREF(dict);
     }
+  error_check:
```

Comment:
There is only one goto to this label. Could this check not be inlined? Or did you mean to jump here from line 1410?

Reply:
It is reachable by failing through from the code above as well.

Line 1410 sets the error, no need to check for it. That's why it jumps to `done`.
I'll admit the code is a bit convoluted, but I think that is indicative of the underlying complexity.